enhancements/insights/container-scanner.md

```diff
@@ -0,0 +1,498 @@
+
+* As a middleware product manager, I can track the number of applications built using Quarkus to understand our return on investment from an OpenShift workload perspective.
+
+* As a RHEL product manager, I can follow trends in the use of other Linux containers to inform my product roadmap.
+
+* As an OpenShift product marketing manager, I want to be able to produce content that shows users use OpenShift as a  platform for all types of workloads.
+    
+
+### Goals
+
+The goal of this proposal is to scan running containers and pop a few key pieces of data to identify the workload that they are running. This data will enrich and complement the information already gathered by the Insights Operator to get a more accurate representation of workloads running on OpenShift.
+
+### Non-Goals
+
+* It is not a goal of this proposal to scan container for security purposes
+* It is not a goal of this proposal to get runtime information specific to a given customer or application
+* It is not a goal of this proposal to capture personal sensitive information
+
+## Proposal
+
+This enhancement proposal introduces additional data to identify the technology stack of running containers.
+
+The existing `workloadContainerShape` type that contains workload information about a container will be enhanced with a new `workloadRuntimeInfoContainer` field:
+
+```go
+// workloadContainerShape describes the shape of a container which includes
+// a subset of the data in the container.
+// TODO: this may desirable to make more precise with a whole container hash
+//
+//  that includes more of the workload, but that would only be necessary if
+//  it assisted reconstruction of type of workloads.
+type workloadContainerShape struct {
+   // ImageID is the content addressable hash of the image as observed from
+   // the status or the spec tag.
+   ImageID string `json:"imageID"`
+   // FirstCommand is a hash of the first value in the command array, if
+   // any was set.
+   FirstCommand string `json:"firstCommand,omitempty"`
+   // FirstArg is a hash of the first value in the arguments array, if any
+   // was set.
+   FirstArg string `json:"firstArg,omitempty"`
+   // Runtime info on the container
+   RuntimeInfo workloadRuntimeInfoContainer `json:"runtimeInfo,omitempty"`
+}
+```
+
+This new `workloadRuntimeInfoContainer` is defined as:
+
+```go
+type workloadRuntimeInfoContainer struct {
+   // Hash of the identifier of the Operating System
+   Os string `json:"os,omitempty"`
+   // Hash of the version identifier of the Operating System
+   OsVersion string `json:"osVersion,omitempty"`
+   // Hash of identifier of the kind of runtime
+   Kind string `json:"kind,omitempty"`
+   // Hash of the version of the kind of runtime
+   KindVersion string `json:"kindVersion,omitempty"`
+   // Hash of the entity that provides the runtime-kind implementation
+   KindImplementer string `json:"kindImplementer,omitempty"`
```

Comment:
Perhaps the `Os` and the `Kind` would deserve their own structs, but it's just a detail.